runtimes/runtimes/fqn/common/fqnExtractor.test.ts

```diff
@@ -0,0 +1,114 @@
+import * as fqn from '@aws/fully-qualified-names'
+import * as assert from 'assert'
+import sinon from 'ts-sinon'
+import { findNames } from './fqnExtractor'
+
+describe('fqnExtractor.findNames', () => {
+    let typescriptStub: sinon.SinonStub
+    let javaStub: sinon.SinonStub
+    const tsResult = {}
+    const javaResult = {}
+    const mockRange = {
+        start: {
+            line: 0,
+            character: 0,
+        },
+        end: {
+            line: 1,
+            character: 1,
+        },
+    }
+    const mockFileText = `console.log('abc')`
+
+    beforeEach(() => {
+        typescriptStub = sinon.stub(fqn.TypeScript, 'findNames').callsFake(() => Promise.resolve(tsResult))
+        javaStub = sinon.stub(fqn.Java, 'findNames').callsFake(() => Promise.resolve(javaResult))
+    })
+
+    afterEach(() => {
+        typescriptStub.restore()
+        javaStub.restore()
+    })
+
+    it('throws error with unsupported languageId', async () => {
+        await assert.rejects(() =>
+            findNames(fqn, {
+                languageId: 'lolcode' as any,
+                fileText: mockFileText,
+            })
+        )
+    })
+
+    it('calls the corresponding fqn function', async () => {
+        let result = await findNames(fqn, {
+            languageId: 'typescript',
+            fileText: mockFileText,
+        })
+
+        sinon.assert.calledOnceWithMatch(typescriptStub, mockFileText)
```

Comment:
why are any of these tests using mocks (sinon)? can't they open an actual text document and check the actual fqn results? there is no network/service calls involved, right?